src/cluster.py

```diff
@@ -710,6 +731,57 @@ def primary_changed(self, old_primary: str) -> bool:
         primary = self.get_primary()
         return primary != old_primary
 
+    def remove_raft_data(self) -> None:
+        """Stops Patroni and removes the raft journals."""
+        logger.info("Stopping patroni")
+        self.stop_patroni()
+
+        logger.info("Wait for postgresql to stop")
+        for attempt in Retrying(wait=wait_fixed(5)):
+            with attempt:
+                for proc in psutil.process_iter(["name"]):
+                    if proc.name() == "postgres":
+                        raise RaftPostgresqlStillUpError()
+
+        logger.info("Removing raft data")
+        try:
+            path = Path(f"{PATRONI_CONF_PATH}/raft")
+            if path.exists() and path.is_dir():
+                shutil.rmtree(path)
+        except OSError as e:
+            raise Exception(f"Failed to remove previous cluster information with error: {str(e)}")
+        logger.info("Raft ready to reinitialise")
+
+    def reinitialise_raft_data(self) -> None:
+        """Reinitialise the raft journals and promoting the unit to leader. Should only be run on sync replicas."""
+        logger.info("Rerendering patroni config without peers")
+        self.charm.update_config(no_peers=True)
+        logger.info("Starting patroni")
+        self.start_patroni()
+
+        logger.info("Waiting for new raft cluster to initialise")
+        for attempt in Retrying(wait=wait_fixed(5)):
+            with attempt:
```

Comment:
We have to define sane timeout for the cases when the recovery hijacks execution.